hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -129,7 +129,7 @@ case class HoodieFileIndex(
 
     // To support metadata listing via Spark SQL we allow users to pass the config via SQL Conf in spark session. Users
     // would be able to run SET hoodie.metadata.enable=true in the spark sql session to enable metadata listing.
-    properties.put(HoodieMetadataConfig.ENABLE,
+    properties.setProperty(HoodieMetadataConfig.ENABLE.key(),
```

Comment:
Previously we were putting the object, and now it is prop string, otherwise both `put()/setProperty()`  should work right?

Reply:
we always look for string keys, thats why.